### PR TITLE
CT-1573: show video status after user upload

### PR DESCRIPTION
**Pull request by omid**

### CODE REVIEW

File: lib/ApiClient/AbstractCoreApiClient.php [hunk 1]
```diff
@@ -217,29 +183,16 @@ protected function buildJsonParameters(array $required, array $optional)
...
      * Generates the cache key based on the class name, request method, uri and options.
-     *
-     * @param string $method
-     * @param string $uri
-     * @param array  $options
-     *
-     * @return string
      */
-    private function generateCacheKey($method, $uri, array $options = [])
+    private function generateCacheKey(string $method, string $uri, ?array $options = []): string
```
Comment (robertszeker) [inline lib/ApiClient/AbstractCoreApiClient.php]: Is there a reason, you make options nullable? default can stay as empty array.
```suggestion
    private function generateCacheKey(string $method, string $uri, array $options = []): string
```

File: lib/Entity/ApiCredentials.php [hunk 2]
```diff
@@ -19,30 +16,18 @@ class ApiCredentials
...
     private $password;
 
-    /**
-     * ApiCredentials constructor.
-     *
-     * @param string $username
-     * @param string $password
-     */
-    public function __construct($username, $password)
+    public function __construct(?string $username, ?string $password)
```
Comment (robertszeker) [inline lib/Entity/ApiCredentials.php:19]: If it's really nullable, then the getters must also return ?string

File: lib/Factory/Guzzle6ApiClientFactory.php [hunk 3]
```diff
@@ -10,50 +12,33 @@
...
      * Method to instantiate a HTTP client.
-     *
-     * @param string $baseUri
-     * @param array  $middlewares
-     * @param array  $options
-     *
-     * @return ClientInterface
      */
-    public function createHttpClient($baseUri, array $middlewares = [], array $options = [])
+    public function createHttpClient(string $baseUri, ?array $middlewares = [], ?array $options = []): ClientInterface
```
Comment (robertszeker) [inline lib/Factory/Guzzle6ApiClientFactory.php]: null as $middlewares will cause a big fat WARNING in the foreach loop

File: lib/Interfaces/ApiClientInterface.php [hunk 4]
```diff
@@ -2,286 +2,193 @@
...
-     * @param string $embedType
-     *
-     * @return EmbedCode
      */
-    public function getEmbedCode($videoManagerId, $videoId, $playerDefinitionId, $embedType = 'html');
+    public function getEmbedCode(
+        int $videoManagerId,
+        string $videoId,
+        string $playerDefinitionId,
+        ?string $embedType = 'html'
```
Comment (robertszeker) [inline lib/Interfaces/ApiClientInterface.php]: It wasn't nullable in the doc block

File: lib/Interfaces/ApiClientInterface.php [hunk 5]
```diff
@@ -2,286 +2,193 @@
...
-     * @param $timestamp
-     *
-     * @return ThumbnailInterface
-     */
-    public function createThumbnailByTimestamp($videoManagerId, $videoId, $timestamp);
+    public function createThumbnailByTimestamp(
+        int $videoManagerId,
+        string $videoId,
+        int $timestamp
+    ): ?ThumbnailInterface;
```
Comment (robertszeker) [inline lib/Interfaces/ApiClientInterface.php:182]: Why is the return value nullable?

**Review by kudrmudr — APPROVED** [review]

File: lib/Interfaces/ApiClientInterface.php [hunk 6]
```diff
@@ -2,286 +2,193 @@
...
+    ): string;
 
     /**
      * Get list of videos.
      *
-     * @param int                     $videoManagerId
-     * @param VideosRequestParameters $parameters
-     *
-     * @return VideoInterface[] Collection of videos
+     * @return ArrayCollection<<VideoInterface> Collection of videos
```
Comment (robertszeker) [inline lib/Interfaces/ApiClientInterface.php]:
```suggestion
     * @return ArrayCollection<VideoInterface> Collection of videos
```

File: lib/Interfaces/ApiClientInterface.php (hunk 5 above)
Comment (omid) [reply lib/Interfaces/ApiClientInterface.php:182]: Because in the only implementation we return null in one case. I didn't change the logic there.

**Review by robertszeker — APPROVED** [review]

**Review by kudrmudr — APPROVED** [review]

**Pull request merged**